trade-finance-manager-ui/server/controllers/utilisation-reports/find-reports-by-year/search-input-validator.ts

```diff
@@ -0,0 +1,60 @@
+import { isNonEmptyString } from '@ukef/dtfs2-common';
+import { REGEX } from '../../../constants';
+
+type Error = {
+  text: string;
+  href: string;
+};
+
+export type FindUtilisationReportByYearValidationErrors = {
+  errorSummary: Error[];
+  bankError: string | undefined;
+  yearError: string | undefined;
+};
+
+const isBefore2024 = (year: number): boolean => year < 2024;
+
+const isInTheFuture = (year: number): boolean => year > new Date().getFullYear();
+
+const validateYearInputAndReturnErrorMessageIfInvalid = (year: unknown): string | undefined => {
+  const validYearRegex = new RegExp(REGEX.YEAR);
```

Comment:
This creation of a `RegExp` object isn't needed - a regex literal can be used in the same was as a `RegExp` object: `REGEX.YEAR.test(year)`